src/quantum_object.jl

```diff
@@ -717,6 +717,110 @@ Quantum Object:   type=Operator   dims=[20, 20]   size=(400, 400)   ishermitian=
 """
 ⊗(A::QuantumObject, B::QuantumObject) = kron(A, B)
 
+@doc raw"""
+    permute(A::QuantumObject, order::Vector{Int})
+
+Permute the subsystems of a [`QuantumObject`](@ref) `A` according to the order specified in `order`. So, if `order = [2, 1, 3]`, the first subsystem will become the second, the second will become the first, and the third will remain the same, i.e. H₁ ⊗ H₂ ⊗ H₃ → H₂ ⊗ H₁ ⊗ H₃.
+
+This method currently dispatches only for [`Ket`](@ref), [`Bra`](@ref), and [`Operator`](@ref) types of [`QuantumObject`](@ref).
+
+# Examples
+
+```julia
+ψ1 = fock(2, 0)
+ψ2 = fock(2, 1)
+ψ_12 = tensor(ψ1, ψ2)
+
+permute(ψ_12, [2, 1]) ≈ tensor(ψ2, ψ1)
+
+# Returns true
+```
+
+"""
+function permute end
```

Comment:
Maybe just put the definition of functions here, we can merge those three methods in just one line, namely
```julia
function permute(
    A::QuantumObject{<:AbstractArray{T},ObjType},
    order::AbstractVector{Int}
) where {T,ObjType<:Union{KetQuantumObject,BraQuantumObject,OperatorQuantumObject}}
    if length(order) != length(A.dims)
        throw(ArgumentError("The order vector must have the same length as the number of subsystems"))
    end
    perm = kron_permutation(order, op.dims)
    return QuantumObject(A.data[perm, perm], A.type, A.dims[order])
end
```